docs/proposals/push-metrics.md

```diff
@@ -0,0 +1,69 @@
+# Heapster Push Metrics
+
+## Overview and Motivation
+
+Currently, Heapster supports pulling metrics from kubelet, and defines an
+interface for pulling from other sources.  However, in certain cases, it is
+more useful to be able to have services push metrics into Heapster, instead
+of having Heapster pull metrics.
+
+For instance, supporting push metrics makes it easy for cluster admins to add
+custom metrics into Heapster with relatively minimal effort: they can simply
+write a program or script which collects and processes the information, and
+then add a recurring Job or cron job which pushes the metrics (instead of
```

Comment:
This seems like a nice approach to me. It would mean no change to the current flow of data into Heapster, no requirement for authentication changes, no flow control to prevent overloading Heapster.